pyanaconda/modules/localization/installation.py

```diff
@@ -58,3 +70,139 @@ def _write_language_configuration(self, lang, root):
+
+    def __init__(self, sysroot, x_layouts, switch_options, vc_keymap):
+        """Create a new task,
+
+        :param sysroot: a path to the root of the installed system
+        :type sysroot: str
+        :param x_layouts: list of x layout specifications
+        :type x_layouts: list(str)
+        :param switch_options: list of options for layout switching
+        :type switch_options: list(str)
+        :param vc_keymap: virtual console keyboard mapping name
+        :type vc_keymap: str
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._x_layouts = x_layouts
+        self._switch_options = switch_options
+        self._vc_keymap = vc_keymap
+
+    @property
+    def name(self):
+        return "Configure keyboard"
+
+    def run(self):
+        if self._x_layouts:
+            write_x_configuration(
+                self._x_layouts,
+                self._switch_options,
+                self._sysroot
+            )
+        if self._vc_keymap:
+            write_vc_configuration(
+                self._vc_keymap,
+                self._sysroot
+            )
+
+
+def write_x_configuration(x_layouts, switch_options, root):
+    """Write X keyboard layout configuration to the configuration files.
+
+    :param x_layouts: list of x layout specifications
+    :type x_layouts: list(str)
+    :param switch_options: list of options for layout switching
+    :type switch_options: list(str)
+    :param root: root path of the configured system
+    :type root: str
+    """
+
+    localed_wrapper = LocaledWrapper()
+    xconf_file_path = os.path.normpath(X_CONF_DIR + "/" + X_CONF_FILE_NAME)
+    errors = []
+
+    try:
+        if not os.path.isdir(X_CONF_DIR):
+            os.makedirs(X_CONF_DIR)
+    except OSError:
+        errors.append("Cannot create directory {}".format(X_CONF_DIR))
+
+    if root != "/":
+        # writing to a different root, we need to save these values, so that
+        # we can restore them when we have the file written out
+        layouts_variants = localed_wrapper.layouts_variants
+        options = localed_wrapper.options
+
+        # set systemd-localed's layouts, variants and switch options, which
+        # also generates a new conf file
+        localed_wrapper.set_layouts(x_layouts,
+                                    switch_options)
+
+        # make sure the right directory exists under the given root
+        rooted_xconf_dir = os.path.normpath(root + "/" + X_CONF_DIR)
+        try:
+            if not os.path.isdir(rooted_xconf_dir):
+                os.makedirs(rooted_xconf_dir)
+        except OSError:
+            errors.append("Cannot create directory {}".format(rooted_xconf_dir))
+
+        # copy the file to the chroot
+        try:
+            shutil.copy2(xconf_file_path,
+                            os.path.normpath(root + "/" + xconf_file_path))
+        except IOError:
+            # The file may not exist (eg. text install) so don't raise
```

Comment:
Don't you think it could be helpful to have debug output. I think it could be handy to know that something is missing even though it's not required.

Reply:
Added debug message